### ISSUE TITLE
plugins: support passing environment variables for the custom command

### ISSUE THREAD

**Issue by lzakharov** [golangci/golangci-lint#4703]
### Your feature request related to a problem? Please describe

I want to build versions of `golangci-lint` with custom linters for specific platforms/architectures, which may be different from the host one. 

### Describe the solution you'd like

It would be nice to have an ability to pass custom environment variables when building a new version of `golangci-lint`:

```sh
golangci-lint custom -E GOOS=linux -E GOARCH=amd64
```

Environment injection should be done somewhere here: https://github.com/golangci/golangci-lint/blob/v1.58.0/pkg/commands/internal/builder.go#L177-L186.

### Describe alternatives you've considered

Currently, "cross compilation" can be done using platform/architecture specific containers, which is IMHO not the most convenient option.

Another solution is to support a configurable list of targets in the `custom-gcl`, but I find this solution less elegant. 

### Additional context

If this proposal is valid, I'll be happy to prepare a pull request.

**Comment by ldez** [comment]
Hello,

in which context do you need to create a binary for a different arch than the current OS?

**Comment by ldez** [comment]
I think an option is not needed, because you can do that:

```bash
GOOS=windows GOARCH=amd64 golangci-lint custom
```

**Comment by lzakharov** [comment]
Oh, that's right. For some reason, I decided that the environment from the current process would not be used when executing the command. Thanks for clarification!
